igvm/host.py

```diff
@@ -134,3 +136,23 @@ def put(self, remote_path, local_path, mode='0644'):
                 'mv {0} {1} ; chmod {2} {1}'
                 .format(tempfile, remote_path, mode)
             )
+
+    def lock(self, allow_fail=False):
+        if self.dataset_obj['igvm_locked']:
+            raise InvalidStateError(
+                'Server "{0}" is already being worked on by another igvm'
+                .format(self.dataset_obj['hostname'])
+            )
+
+        self.dataset_obj['igvm_locked'] = True
+        try:
+            self.dataset_obj.commit()
+        except DatasetError:
+            raise InvalidStateError(
+                'Server "{0}" is already being worked on by another igvm'
+                .format(self.dataset_obj['hostname'])
+            )
+
+    def unlock(self):
```

Comment:
Maybe `aquire_lock()` and `release_lock()` are better names to emphasis state-changing behavior of the methods.

Reply:
Sure, done.